xrpl4j-address-codec/src/test/java/org/xrpl/xrpl4j/codec/addresses/AddressTest.java

```diff
@@ -253,4 +253,51 @@ public void xAddressWith64BitTag() {
     );
   }
 
+
+  @Test
+  public void addressWithBadChecksum() {
+    Address address = Address.of("r9cZA1mLK5R5am25ArfXFmqgNwjZgnfk59");
+
+    assertThrows(
+      EncodingFormatException.class,
+      () -> addressCodec.classicAddressToXAddress(address, true),
+      "Checksum does not validate"
+    );
+  }
```

Comment:
This test should go in `AddressCodecTest`. The above XAddress tests should also probably go in there. Can you create a ticket to move those over?